Refresh.GameServer/Endpoints/Game/Levels/PublishEndpoints.cs

```diff
@@ -13,10 +15,57 @@ namespace Refresh.GameServer.Endpoints.Game.Levels;
 
 public class PublishEndpoints : EndpointGroup
 {
+    /// <summary>
+    /// Does basic verification on a level
+    /// </summary>
+    /// <param name="body">The level to verify</param>
+    /// <param name="user">The user that is attempting to upload</param>
+    /// <param name="logger">A logger instance</param>
+    /// <returns>Whether or not validation succeeded</returns>
+    private static bool VerifyLevel(GameLevelRequest body, GameUser user, LoggerContainer<BunkumContext> logger)
+    {
+        if (body.Title.Length > 256)
+        {
+            return false;
+        }
+
+        if (body.Description.Length > 4096)
+        {
+            return false;
+        }
+
+        if (body.MaxPlayers > 4 || body.MinPlayers > 4)
+        {
+            return false;
+        }
+
+        if (body.TeamPicked)
+        {
+            logger.LogWarning(BunkumContext.UserContent, $"User {user.Username} attempted to force their level to be team picked! This is very likely a forged request.");
+            return false;
+        }
+
+        if (body.BooCount != 0 || body.YayCount != 0 || body.HeartCount != 0)
+        {
+            logger.LogWarning(BunkumContext.UserContent, $"User {user.Username} attempted to force non-0 boo/yay/heart counts! This is very likely a forged request.");
+            return false;
+        }
+
+        if (body.PlayerCount > 4)
+        {
+            return false;
+        }
```

Comment:
Not sure if it's worth logging/checking these

Reply:
Yeah it may just be worth removing those fields from the GameLevelRequest object altogether, since they dont *need* to exist for requests, mostly they are there to catch any attempted request forgery that someone may try

Reply:
At the end of the day I think it's up to instance moderators to handle request forgery, but we absolutely should block it where we should. In this case, I don't like having these checks because:

- The forgery wouldn't work anyways due to the nature of how Refresh handles Levels
- It creates a mess in the Request class
- The Request class is meant to silently drop things like this in the first place

Reply:
Plus, having the extra properties could lead to confusion for developers